bazel/envoy_test.bzl

```diff
@@ -155,35 +156,41 @@ def envoy_cc_test(
         coverage = True,
         local = False,
         size = "medium"):
-    test_lib_tags = []
     if coverage:
-        test_lib_tags.append("coverage_test_lib")
+        coverage_tags = tags + ["coverage_test_lib"]
+    else:
+        coverage_tags = tags
     _envoy_cc_test_infrastructure_library(
         name = name + "_lib_internal_only",
         srcs = srcs,
         data = data,
         external_deps = external_deps,
         deps = deps + [repository + "//test/test_common:printers_includes"],
         repository = repository,
-        tags = test_lib_tags,
+        tags = coverage_tags,
         copts = copts,
         # Allow public visibility so these can be consumed in coverage tests in external projects.
         visibility = ["//visibility:public"],
     )
+    if coverage:
+        coverage_tags = tags + ["coverage_test"]
     native.cc_test(
         name = name,
         copts = envoy_copts(repository, test = True) + copts,
         linkopts = _envoy_test_linkopts(),
         linkstatic = envoy_linkstatic(),
         malloc = tcmalloc_external_dep(repository),
-        deps = envoy_stdlib_deps() + [
-            ":" + name + "_lib_internal_only",
-            repository + "//test:main",
-        ],
+        deps = select({
+            "@envoy//bazel:windows_x86_64": [repository + "//test:dummy_main"],
```

Comment:
This is not needed at the end, and shouldn't included in this PR?

Reply:
We saw this exclusion for Apple so we added it for Windows.

We don't have a strong opinion either way and we can remove this change.

Reply:
@achasveachas the exclusion you mentioned is for fuzz I think? No platform should be excluded from normal cc_test.